packages/makeswift/src/integrate-next-app.ts

```diff
@@ -33,6 +37,24 @@ function installMakeswiftRuntime({ dir }: { dir: string }): void {
   }
 }
 
+function addMakeswiftApiRoute({ dir }: { dir: string }): void {
+  const pagesFolder = getPagesFolder({ dir })
+  const extension = getExtension({ dir })
+
+  // If Makeswift API folder does not exist, create
+  createFolderIfNotExists(path.join(pagesFolder, 'api'))
+  createFolderIfNotExists(path.join(pagesFolder, 'api', 'makeswift'))
+
+  const apiRoute = `import { MakeswiftApiHandler } from '@makeswift/runtime/next'
```

Comment:
The contents of the file should be different if it's a TypeScript installation vs plain JavaScript. This is specially marked in the `[[...path]].js` file since there's a lot of types we want to include there. We might want to have a `typescript: boolean` option we can pass for all these functions that generate files so that we can detect TypeScript when running the CLI and then can act based on that.

Reply:
We do have that, but I agree that it should be called once and be a parameter to the functions. I'm going to add that.

Also, do we have an example somewhere of what the TS version should look like, if it's different? I was going by [this](https://www.makeswift.com/docs/guides/manual-setup#add-the-makeswift-api-handler).

Reply:
https://github.com/makeswift/makeswift/blob/07b469e0ed6ad8290d36f42add86526b4047954b/examples/basic-typescript/pages/api/makeswift/%5B...makeswift%5D.ts

```typescript
import { MakeswiftApiHandler } from '@makeswift/runtime/next'

export default MakeswiftApiHandler(process.env.MAKESWIFT_SITE_API_KEY!)
```

Aren't these the same?

Reply:
The one that's mostly different is the dynamic catch-all route. For this one the only different is the non-null assertion in the API key so that TypeScript doesn't error